### PR TITLE
[cmake] update enabled OT options in `openthread/CMakeLists.txt`

### PR DESCRIPTION
**Pull request by abtink**
This commit updates `openthread/CMakeLists.txt`, simplifying how the configs are enabled. The following features are also now explicitly enabled:

- `OT_15_4`,
- `OT_ANYCAST_LOCATOR`,
- `OT_CHILD_SUPERVISION`,
- `OT_DATASET_UPDATER`,
- `OT_DNS_CLIENT`,
- `OT_ECDSA`,
- `OT_HISTORY_TRACKER`,
- `OT_NETDATA_PUBLISHER`,
- `OT_SERVICE`,
- `OT_SRP_CLIENT`,
- `OT_UPTIME`.

### CODE REVIEW

File: third_party/openthread/CMakeLists.txt [hunk 1]
```diff
@@ -26,47 +26,50 @@
...
 set(OT_MAC_FILTER ON CACHE STRING "enable MAC filter" FORCE)
 set(OT_NAT64_BORDER_ROUTING ${OTBR_NAT64} CACHE STRING "enable NAT64 in border routing manager" FORCE)
 set(OT_NAT64_TRANSLATOR ${OTBR_NAT64} CACHE STRING "enable NAT64 translator" FORCE)
+set(OT_NETDATA_PUBLISHER ON CACHE STRING "enable netdata publisher" FORCE)
 set(OT_PLATFORM "posix" CACHE STRING "use posix platform" FORCE)
 set(OT_PLATFORM_NETIF ON CACHE STRING "enable platform netif" FORCE)
 set(OT_PLATFORM_UDP ON CACHE STRING "enable platform UDP" FORCE)
+set(OT_SERVICE ON CACHE STRING "enable service" FORCE)
 set(OT_SLAAC ON CACHE STRING "enable SLAAC" FORCE)
+set(OT_SRP_CLIENT ON CACHE STRING "enable SRP client" FORCE)
```
Comment (bukepo) [inline third_party/openthread/CMakeLists.txt:61]: Same as DNS client, I'm not sure if SRP client is a requirement for a border router.

File: third_party/openthread/CMakeLists.txt [hunk 2]
```diff
@@ -26,47 +26,50 @@
...
 set(OT_BORDER_ROUTING_COUNTERS ${OTBR_BORDER_ROUTING_COUNTERS} CACHE STRING "enable border routing counters feature" FORCE)
 set(OT_BUILD_EXECUTABLES OFF CACHE STRING "disable building executables" FORCE)
 set(OT_BUILTIN_MBEDTLS_MANAGEMENT OFF CACHE STRING "diable mbedTLS management" FORCE)
+set(OT_CHILD_SUPERVISION ON CACHE STRING "enable child supervision" FORCE)
 set(OT_COAP ON CACHE STRING "Enable CoAP in OpenThread")
 set(OT_COAPS ON CACHE STRING "Enable secure CoAP in OpenThread")
 set(OT_COMMISSIONER ON CACHE STRING "enable commissioner")
 set(OT_DAEMON ON CACHE STRING "enable daemon mode" FORCE)
+set(OT_DATASET_UPDATER ON CACHE STRING "enable dataset updater" FORCE)
+set(OT_DNS_CLIENT ON CACHE STRING "enable DNS client" FORCE)
```
Comment (bukepo) [inline third_party/openthread/CMakeLists.txt:45]: I'm not sure if DNS client is a requirement for a border router. Will border router itself directly use this feature?

File: third_party/openthread/CMakeLists.txt (hunk 1 above)
Comment (abtink) [reply third_party/openthread/CMakeLists.txt:61]: They can be useful for testing (having the CLI commands). 

I don't think there is any harm in having them enabled. 
- They add functionality/APIs (and CLI commands) but don't add any run-time overhead if their APIs are not called. 
- And we don't need to worry about code/memory size on posix.

**Review by superwhd — APPROVED** [review]
LGTM :+1:

**Review by bukepo — APPROVED** [review]

File: third_party/openthread/CMakeLists.txt (hunk 1 above)
Comment (bukepo) [reply third_party/openthread/CMakeLists.txt:61]: Just the `FORCE` may indicate the feature is required for a border router, which is not true.

**Review by jwhui — APPROVED** [review]

**Pull request merged**